### PR TITLE
ICU-21165 Produce units.txt resource file (with data from CLDR's units.xml)

### PR DESCRIPTION
**Pull request by hugovdm**
Data files produced with:

    ant -f build-icu-data.xml -DoutDir=/tmp/new_dir -DcldrVersion=37 \
    -DoutputTypes=UNITS,SUPPLEMENTAL_DATA

Once good, commits will be squashed and approval (re-)requested for submission. Before that, only new commits added to aid code review. That's probably best-practice under just about all circumstances, right?

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21165
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: tools/cldr/cldr-to-icu/src/main/resources/ldml2icu_supplemental.txt [hunk 1]
```diff
@@ -145,6 +140,69 @@
 //supplementalData/calendarData/calendar[@type="(%W)"]/eras/era[@type="(%W)"][@(start|end)="(%A)"]
     ; /calendarData/$1/eras/$2/$3:intvector ; values=&ymd($4)
 
+# units.xml
+
+//supplementalData/unitConstants/unitConstant[@constant="(%W)"][@value="(%A)"]
+    ; /unitConstants/$1 ; values="$2"
+
+//supplementalData/unitConstants/unitConstant[@constant="(%W)"][@value="(%A)"][@status="(%W)"]
```
Comment (hagbard) [inline tools/cldr/cldr-to-icu/src/main/resources/ldml2icu_supplemental.txt:148]: use (?:...)? for optional attributes you just ignore.
I.e. you can roll up this rule into the parent one.

File: icu4c/source/data/misc/supplementalData.txt [hunk 2]
```diff
@@ -2104,7 +2104,7 @@ supplementalData:table(nofallback){
             "islamic-tbla",
         }
     }
-    cldrVersion{"37"}
+    cldrVersion{"38"}
```
Comment (hagbard) [inline icu4c/source/data/misc/supplementalData.txt:2107]: I don't know if this is wanted. You should be able to force the cldr version from the Ant build file.
There are lots of options there you might want to read up on.

File: tools/cldr/cldr-to-icu/src/main/resources/ldml2icu_supplemental.txt (hunk 1 above)
Comment (hugovdm) [reply tools/cldr/cldr-to-icu/src/main/resources/ldml2icu_supplemental.txt:148]: Done, PTAL?

File: icu4c/source/data/misc/supplementalData.txt (hunk 2 above)
Comment (hugovdm) [reply icu4c/source/data/misc/supplementalData.txt]: -DcldrVersion=37 cleaned this up, thanks!

File: icu4c/source/data/misc/supplementalData.txt [hunk 3]
```diff
@@ -35765,525 +35765,6 @@ supplementalData:table(nofallback){
             preferred{"H"}
         }
     }
-    unitPreferenceData{
```
[the remark this answers is not in the document]
Comment (hugovdm) [reply icu4c/source/data/misc/supplementalData.txt:35768]: Generated data: tests pass, so I'm not too worried about (re)moving of data, and the new data produced is already working well for us (dev branch).

File: tools/cldr/cldr-to-icu/src/main/resources/ldml2icu_supplemental.txt [hunk 4]
```diff
@@ -145,6 +140,49 @@
 //supplementalData/calendarData/calendar[@type="(%W)"]/eras/era[@type="(%W)"][@(start|end)="(%A)"]
     ; /calendarData/$1/eras/$2/$3:intvector ; values=&ymd($4)
 
+# units.xml
+
+//supplementalData/unitConstants/unitConstant[@constant="(%W)"][@value="(%A)"](?:[@status="(?:%W)"])?
+    ; /unitConstants/$1 ; values="$2"
+
+//supplementalData/unitQuantities/unitQuantity[@baseUnit="(%W)"][@quantity="(%W)"](?:[@status="(?:%W)"])?
```
[the remark this answers is not in the document]
Comment (hugovdm) [reply tools/cldr/cldr-to-icu/src/main/resources/ldml2icu_supplemental.txt:148]: No worries, doesn't bother me: I feel a (readable) little bit of repetition is way better than over-complication due to excessive application of "DRY" principles. ;-)

File: tools/cldr/cldr-to-icu/src/main/resources/ldml2icu_supplemental.txt [hunk 5]
```diff
@@ -145,6 +140,49 @@
 //supplementalData/calendarData/calendar[@type="(%W)"]/eras/era[@type="(%W)"][@(start|end)="(%A)"]
     ; /calendarData/$1/eras/$2/$3:intvector ; values=&ymd($4)
 
+# units.xml
+
+//supplementalData/unitConstants/unitConstant[@constant="(%W)"][@value="(%A)"](?:[@status="(?:%W)"])?
```
[the remark this answers is not in the document]
Comment (hugovdm) [reply tools/cldr/cldr-to-icu/src/main/resources/ldml2icu_supplemental.txt]: Done. Thanks!

**Comment by hugovdm** [comment]
Thanks, squashing!

**Pull request merged**